src/rgw/rgw_data_sync.cc

```diff
@@ -1042,7 +1042,7 @@ class RGWDataSyncSingleEntryCR : public RGWCoroutine {
         if (retcode < 0) {
           tn->log(0, SSTR("ERROR: failed to log sync failure: retcode=" << retcode));
         }
-        if (error_repo && !error_repo->append(raw_key)) {
+        if (!error_repo || !error_repo->append(raw_key)) {
```

Comment:
`RGWDataSyncShardCR::full_sync()` will be spawning `RGWDataSyncSingleEntryCR` with `error_repo=nullptr` - in that case, this error message would be misleading. so i think this line should remain as is

Reply:
If the bucket shards sync fails and error_repo=nullptr, it won't sync anymore. So I think the error message is necessary.
Maybe we should do the same thing to wait all sync done then release the error_repo in full_sync ?

Reply:
if `RGWDataSyncSingleEntryCR` fails, it will already be printing the actual error in the log. i don't think another error here related to error_repo will do anything but confuse the reader

Reply:
this log means RGWDataSyncSingleEntryCR fails and not recorded in error_repo, this should not happen after this fix.  If happens, the log will print and we should find out why.